File: Dataset/Leetcode/train/69/605.java
class Solution {
    public int XXX(int x) {
        int l = 0,r = x,m;
        if(x>46339){
            r=46339;
        }
        while(l<=r){
            // System.out.println(l+" "+r);
            m = l+(r-l)/2;
            if(m*m==x){
                
                return m;
            }else if(m*m>x){
                r=m-1;
            }else{
                l=m+1;
            }
        }
        
        if(l*l>x){
            return l-1;
        }
        return l;
    }
}

File: Dataset/Leetcode/train/46/605.cpp
 class Solution {
public:
    vector<vector<int>> res;
    vector<vector<int>> XXX(vector<int>& nums) {
        //使用dfs
        vector<int> used(nums.size());//初始化一个bool作用的vector，用于标记哪些元素是使用过的
        dfs(nums,used);
        return res;
    }
    vector<int> temp;
    void dfs(vector<int>& nums,vector<int>& used){
        //结束条件
        if(temp.size() == nums.size()) {
            res.push_back(temp);
            return;
        }
        for(int i = 0; i<nums.size(); i++){
            if(!used[i]){
                temp.push_back(nums[i]);
                used[i] = 1;
                dfs(nums, used);
                temp.pop_back();
                used[i] = 0;
            }
        }
    }
};

